scripts/inc.packagetarget.sh

```diff
@@ -94,35 +109,39 @@ createzip() {
         printf "%s\t%s\t%d\n" "$f" "$g" "$foldersize" >> "$build/app_sizes.txt"
       done
 
-      # Wait if we reached RAM or THREADS limit
-      tries=0; while true; do
-        # Count still running createxz instances
-        threads=0; for p in $pidlist; do test -d /proc/$p && threads=$((threads+1)); done
-
-        # Fetch "commitable" memory
-        memory=$(grep "MemAvailable:" /proc/meminfo | awk '{print $2}')
-        echo "Total load: RAM[$memory]/[$MEMORY], CPU[$threads]/[$THREADS], PROC[`for p in $(pgrep xz); do echo -n "$(awk '/VmSize/ { print $2 }' /proc/$p/status) "; done`]"
-
-        # Check if reached our limits (3/4 of RAM or THREADS), wait if we are
-        if [ $threads -ge $THREADS ] || [ $MEMORY -ge $memory ] || [ $memory -lt $MEMORY_MIN ]; then
-          sleep 5
-          # Update tries counter
-          tries=$((tries+1))
-          # If we are trying for more then 180*5 seconds, we bail out (in case machine is to low on memory or CPU we won't run forever!)
-          if [ $tries -gt 180 ]; then
-            echo "Seems like this machine is too slow or was unable to collect enought usable memory for compression, aborting."
-            exit 1
-          fi
-          continue
-        else
-          break
-       fi
-      done
-
-      # Spawn xz creation
-      createxz $d &
-      # Collect resulting PID
-      pidlist="$pidlist $!"
+      # Use parallel mode only if we have memory metric and have more then 1 CPU
+      if [ $THREADS -gt 1 ] && [ $MEMORY -gt 0 ]; then
+        # Wait if we reached RAM or THREADS limit
+        tries=0; while true; do
+          # Count still running createxz instances
+          threads=0; for p in $pidlist; do test -d /proc/$p && threads=$((threads+1)); done
+          memory=$(grep "MemAvailable:" /proc/meminfo | awk '{print $2}')
+
+          # Check if reached our limits (3/4 of RAM or THREADS), wait if we are
+          echo "Resource report: FreeRAM[$memory], InitialFreeRAM[$MEMORY], MinimumFreeRAM[$MEMORY_MIN], Threads[$threads], CPUs[$THREADS], XZ RAM[`for p in $(pgrep xz); do echo -n "$(awk '/VmSize/ { print $2 }' /proc/$p/status) "; done`]"
```

Comment:
I understand that better UI would need e.g. curses.
But if instead of reporting the wait, we could report the start of a new thread, and e.g. passing an optional string parameter to the createxz function, that than print the details within the current output.